polkadot/runtime/parachains/src/assigner_on_demand/mod.rs

```diff
@@ -324,6 +132,21 @@ pub mod pallet {
 		/// The default value for the spot traffic multiplier.
 		#[pallet::constant]
 		type TrafficDefaultValue: Get<FixedU128>;
+
+		/// The maximum number of blocks some historical revenue
+		/// information stored for.
+		#[pallet::constant]
+		type MaxHistoricalRevenue: Get<u32>;
+
+		/// Identifier for the internal revenue balance.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
```

Comment:
That's a fair point, but I see the issue is still open and has some ongoing discussion. It's how it is implemented in many places (I do agree it's nasty), and here, beyond establishing the account address from the relay chain side, it's also used from the parachain side to give Coretime an idea of which account it should teleport funds to. If you have ideas on how to improve it quickly, please suggest. Otherwise, it just works.

Reply:
> it's also used from the parachain side to give Coretime an idea of which account it should teleport funds to.

Then just make it an account directly, instead of passing a `PalletId`.

Reply:
You mean hardcode some `[u8; 32]` there? Sounds even more hacky :/

Reply:
No, but let's keep the `PalletId`...